app/src/main/java/me/quadphase/qpdex/DetailedPokemonActivity.java

```diff
@@ -31,39 +141,453 @@
+     * information is being displayed at any given iteration on a pokemon
+     */
+    private void retrieveInterfaceElements(){
+        pkmnName = (TextView) findViewById(R.id.textview_pkmnname_detail);
+        pkmnSprite = (ImageView) findViewById(R.id.imgbutton_pkmnsprite_detail);
+        pkmnType1 = (ImageView) findViewById(R.id.imgview_pkmntype1_detail);
+        pkmnType2 = (ImageView) findViewById(R.id.imgview_pkmntype2_detail);
+        pkmnDescription = (TextView) findViewById(R.id.textview_detailedpkmndescript);
+        eggGroupSteps = (TextView) findViewById(R.id.textview_eggsteps);
+
+        typeStrong = (LinearLayout) findViewById(R.id.linlayout_typestrong);
+        typeWeak = (LinearLayout) findViewById(R.id.linlayout_typeweak);
+        evolutionChain = (LinearLayout) findViewById(R.id.linlayout_evolutions_detail);
+        eggGroupBox = (LinearLayout) findViewById(R.id.linlay_egggroupbox);
+        abilitiesBox = (LinearLayout) findViewById(R.id.linlay_abilitiesbox);
+        movesBox = (LinearLayout) findViewById(R.id.linlay_movebox);
+
+        evolutionTab = (TextView) findViewById(R.id.title_evolutions);
+        alternatesTab = (TextView) findViewById(R.id.title_altforms);
+    }
+
+    /**
+     * Creates a View element based on "custom_typematch_block" and does all the necessary screen
+     * screen calculations for correct grid alignment. Primarily used with {@see fillTypeComparisonInfo()}
+     * @param quantifier How detailedPokemon is affected by the Types being passed (i.e. Immune, etc)
+     * @param types The list of types that correspond to the Identifier
+     * @return A LinearLayout that can be directly set as a child of any LayoutView
+     */
+    private LinearLayout createTypeMatchBlock(String quantifier, List<Type> types){
+        LinearLayout typeMatchBlock = (LinearLayout) getLayoutInflater().inflate(R.layout.custom_typematch_block, null);
+        GridLayout frame=(GridLayout) typeMatchBlock.findViewById(R.id.gridlay_types);
+        TextView title = (TextView) typeMatchBlock.findViewById(R.id.characteristic);
+
+        title.setText(quantifier);
+
+        for (int i=0;i<types.size();i++) {
+            ImageView typeBadge = new ImageView(this);
+
+            typeBadge.setImageDrawable(
+                    new BitmapDrawable(getResources(),
+                            PokedexAssetFactory.getTypeBadge(this, types.get(i).getName())));
+
+            typeBadge.setLayoutParams(new LinearLayout.LayoutParams(
+                    ViewGroup.LayoutParams.MATCH_PARENT,
+                    ViewGroup.LayoutParams.MATCH_PARENT));
+            typeBadge.setScaleType(ImageView.ScaleType.FIT_XY);
+//            typeBadge.setMaxHeight(30);
+            typeBadge.setAdjustViewBounds(true);
+            typeBadge.getLayoutParams().width = 160;
+            frame.addView(typeBadge);
+        }
+
+        //TODO: We need to calculate for screen sizes
```

Comment:
Haven't tried building on a Tablet yet, but I know they have specific folders for that layout code. In this one the calculation is for correctly scaling the Type sprites into the GridView